src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala

```diff
@@ -0,0 +1,206 @@
+	        for (i <- 0 until lons1.length) { lonData.set(i, lons1(i))}
+	   
+	        datafile.create()
+	        datafile.write(rr, varData)
+	        datafile.write(latDimVar, latData)
+	        datafile.write(lonDimVar, lonData)
+	       
+	        datafile.close()
+	    }
+	    catch{
+	    	case _: Throwable => logger.info("Error generating netCDF file for " + filename + "\n")
+	    }
+	}
+
+	/** Get the data from the CE to write
+	 * @param edgeArray Array[(String, String)] representing the edgelist 
+	 * @param MCSNodeMap mutable.HashMap[String, MCCNode] representing the map of each node metadata
+	 * @param lats Array[Double] representing the latitudes
+	 * @param lons Array[Double] representing the longitudes
+	 * @param tightestBox Boolean to use tightest box around data.
+	 * 
+	 */
+	def get_node_data(edgeArray:Array[(String, String)], MCSNodeMap:mutable.HashMap[String, MCCNode], lats:Array[Double], lons:Array[Double], tightestBox:Boolean ): Unit = {
+		edgeArray.map({ e => 
+		    // import ucar.ma2.{Array, ArrayDouble, DataType}
+
+		    var seenNodes = new ListBuffer[String]()
+		    
+		    val first = e._1.split("CE")(0).split("F")(1)+","+e._1.split("CE")(1).dropRight(1) 
+		    if (!seenNodes.contains(first)){
+		        seenNodes += first
+		        val fNode = MCSNodeMap(first)
+		        extract_masked_data(fNode, lats, lons, tightestBox) 
+		    }
+		    
+		    val second = e._2.split("CE")(0).split("F")(1)+","+e._2.split("CE")(1).dropRight(1)
+		    if (!seenNodes.contains(second)){
+		        seenNodes += second
+		        val sNode = MCSNodeMap(second)
+		        extract_masked_data(sNode, lats, lons, tightestBox)  
+		    }  
+		})
+	}
+
+	/** Get the data from the nodes 
+	 * @param edgeArray Array[(String, String)] representing the edgelist 
+	 * @param MCSNodeMap mutable.HashMap[String, MCCNode] representing the map of each node metadata
+	 * @param lats Array[Double] representing the latitudes
+	 * @param lons Array[Double] representing the longitudes
+	 * @param tightestBox Boolean to use tightest box around data.
+	 * 
+	 */
+	def get_all_node_data(edgeArray:Array[(String, String)], MCSNodeMap:mutable.HashMap[String, MCCNode], lats:Array[Double], lons:Array[Double], tightestBox:Boolean ): Unit = {
+		edgeArray.map({ e => 
+		    
+		    var seenNodes = new ListBuffer[String]()
+		    
+		    val first = e._1.split("CE")(0).split("F")(1)+","+e._1.split("CE")(1).dropRight(1) 
+		    if (!seenNodes.contains(first)){
+		        seenNodes += first
+		        val fNode = MCSNodeMap(first)
+		        extract_masked_data(fNode, lats, lons, tightestBox) 
+		    }
+		    
+		    val second = e._2.split("CE")(0).split("F")(1)+","+e._2.split("CE")(1).dropRight(1)
+		    if (!seenNodes.contains(second)){
+		        seenNodes += second
+		        val sNode = MCSNodeMap(second)
+		        extract_masked_data(sNode, lats, lons, tightestBox)  
+		    }  
+		})
+	}
+
+
+	/** Extract the node mask from the MCCNode metadata for the 
```

Comment:
The style checker uses javadoc style.
That is the the multiline comment asterisk is in line with the first asterisk at the top of the comment block.

For example

/**
 * dadada
 * dadda
 */

instead of 

/**
  * dada
  * dada
  * dada
  */